Mathlib/RingTheory/PowerSeries/Basic.lean

```diff
@@ -853,11 +855,16 @@ theorem coeff_inv_aux [DecidableEq σ] (n : σ →₀ ℕ) (a : R) (φ : MvPower
       if n = 0 then a
       else
         -a *
-          ∑ x in n.antidiagonal, if x.2 < n then coeff R x.1 φ * coeff R x.2 (inv.aux a φ) else 0 :=
-  show inv.aux a φ n = _ by
-    rw [inv.aux]
-    -- unify `Decidable` instances
-    convert rfl
+          ∑ x in antidiagonal n,
+            if x.2 < n then coeff R x.1 φ * coeff R x.2 (inv.aux a φ) else 0 := by
+  change inv.aux a φ n = _
+  nth_rewrite 1 [inv.aux]
+  split_ifs
+  · rfl
+  · apply congr_arg₂ _ rfl
+    apply Finset.sum_congr
+    rw [Subsingleton.elim (Classical.decEq σ) ‹DecidableEq σ›]
+    intros; simp only [dite_eq_ite]; rfl
```

Comment:
I suspect this might not be needed any more

Reply:
Well, `rfl` doesn't work for proving this lemma.

Reply:
While it is true that after the `change` and `rw`, `cong!` concludes (but relatively slowly).